### PR TITLE
Fix build dependencies.

### PR DESCRIPTION
**Pull request by mkarolin**
1. Add brave/components/services/brave_content_browser_overlay_manifest.cc
   dependencies on the manifests of services it includes.

2. Add a patch for chrome/app:generated_resources to include dependency on
   brave/app:brave_generated_resources_grit. Needed because we overwrite
   generated grit/generated_resources.h in chromium_src and include our
   generated grit/brave_generated_resources.h into it.

Fixes brave/brave-browser#4063

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
N/A

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by bsclifton** [comment]
@mkarolin I was able to build this locally - but CI is having some issues. Might be because of the 74 bump - can you check it out?

File: patches/chrome-VERSION.patch [hunk 1]
```diff
@@ -1,5 +1,5 @@
 diff --git a/chrome/VERSION b/chrome/VERSION
-index 3c26e3145cfbeb4928c98357912c90b7b6c4f669..50bdeeab1c597f55ad35713bface77f2a5d832cc 100644
+index 3c26e3145cfbeb4928c98357912c90b7b6c4f669..8e38e4bd7d99ebe81c1b833c50df2dd3bcce0236 100644
 --- a/chrome/VERSION
```
Comment (yrliou) [inline patches/chrome-VERSION.patch]: This change seems unrelated and unnecessary?

Comment (mkarolin) [reply patches/chrome-VERSION.patch]: Yes, looks like it sneaked in while rebasing/fixing CI. Removed, will keep an eye on the CI build.

**Comment by mkarolin** [comment]
CI build passed (https://staging.ci.brave.com/job/brave-browser-build-pr/job/maxk-fix-ci-build-deps/14/).
